cmd/notary-server/config.go

```diff
@@ -278,3 +282,31 @@ func parseServerConfig(configFilePath string, hRegister healthRegister) (context
 		ConsistentCacheControlConfig: consistentCache,
 	}, nil
 }
+
+func setupSignalTrap() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, notary.NotarySupportedSignals...)
+	go func() {
+		for {
+			signalHandle(<-c)
+		}
+	}()
+}
+
+// signalHandle will increase/descrease the logging level via the signal we get.
+func signalHandle(sig os.Signal) {
+	switch sig {
+	case syscall.SIGUSR1:
+		if err := utils.AdjustLogLevel(true); err != nil {
+			fmt.Printf("Attempt to increase log level failed, will remain at %s level, error: %s", logrus.GetLevel(), err)
```

Comment:
Also, do we want to add a newline to the `Printf`?  I noticed when sending signals it didn't seem to log anything until the next successful setting, and then I saw logs like:

```
Attempt to increase log level failed, will remain at debug level, error: log level can not be set higher than DebugAttempt to increase log level failed, will remain at debug level, error: log level can not be set higher than DebugAttempt to increase log level failed, will remain at debug level, error: log level can not be set higher than DebugSuccessfully setting log level to  info
```